nox/tasks.py

```diff
@@ -199,28 +205,32 @@ def filter_manifest(
         # (which is an error condition later).
         manifest.filter_by_keywords(global_config.keywords)
 
-    # Return the modified manifest.
-    return manifest
+    if not manifest:
+        logger.error("No sessions selected after filtering by keyword.")
+        return 3
 
+    if (
+        not global_config.keywords
+        and not global_config.pythons
+        and not global_config.sessions
+        and not global_config.list_sessions
+        and global_config.sessions is not None
+    ):
```

Comment:
I feel the intent could be expressed a little more clearly here. How do you feel about the suggestion below?

I omitted the checks for keywords and pythons because these conditions would have led to bailing out earlier.

```suggestion
    # List sessions if `nox.options.sessions` is an empty sequence.
    if (
        global_config.sessions is not None
        and not global_config.sessions
        and not global_config.list_sessions
    ):
```

Reply:
I redid this, but I kept the keyword/python filtering, since `nox -k keyword` should select and run the tests with `keyword`, rather than forcing it to _also_ be passed via `-s`, I think.

Reply:
See https://github.com/theacodes/nox/pull/523#issuecomment-1001499599